x-pack/plugin/eql/qa/correctness/src/javaRestTest/java/org/elasticsearch/xpack/eql/EsEQLCorrectnessIT.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import com.carrotsearch.randomizedtesting.annotations.TimeoutSuite;
+import org.apache.http.HttpHost;
+import org.apache.lucene.util.TimeUnits;
+import org.elasticsearch.action.admin.cluster.repositories.put.PutRepositoryRequest;
+import org.elasticsearch.action.admin.cluster.snapshots.restore.RestoreSnapshotRequest;
+import org.elasticsearch.client.HttpAsyncResponseConsumerFactory;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.eql.EqlSearchRequest;
+import org.elasticsearch.client.eql.EqlSearchResponse;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.xpack.ql.TestUtils.assertNoSearchContexts;
+
+@TimeoutSuite(millis = 60 * TimeUnits.MINUTE)
+public class EsEQLCorrectnessIT extends ESRestTestCase {
+
+    private static final String PARAM_FORMATTING = "%1$s";
+    private static final String QUERIES_FILENAME = "queries.toml";
+    private static final String INDEX_NAME = "mitre";
+    private static final int FETCH_SIZE = 10000;
+    private static final String GCS_REPO_NAME = "eql_correctness_gcs_repo";
+    private static final String SNAPSHOT_NAME = "mitre-snapshot";
+    private static final String GCS_BUCKET_NAME = "matriv-gcs";
+    private static final String GCS_BASE_PATH = "mitre-data";
+    private static final String GCS_CLIENT_NAME = "eql_test";
```

Comment:
These can be externalized into a properties file.
The "matriv-gcs" should be changed to something more generic.

Reply:
I'd be meaning to do it, but let's get this in as is, and I will rename the bucket soon.